### PR TITLE
Driver-only EC JPAKE: starter

### LINKED ISSUE #7255: Driver-only EC JPAKE: starter
### Description
This creates / updates the infrastructure for testing work on driver-only EC JPAKE.

Context: see https://github.com/Mbed-TLS/mbedtls/issues/6839.
This is about the TL-ecjpake intermediate goal - not doing the work itself, just putting the test components in place. TL-ecjpake will be achieved when the all.sh component introduced here is complete (full config without undue exclusions, running ssl-opt.sh too) and has testing parity with its reference component (that is, no undue "ignore"s in analyze_outcome.py).

Follow-ups: TBD

### PR DESCRIPTION
**Pull request by valeriosetti**
This is the starting PR for testing the parity between non-accelerated and accelerated EC-JPAKE. It follows the same pattern of #6855 and #7142:

- add testing functions in `all.sh`
- extends `analyze_outcomes.py` for verifying that the two implementations cover the same tests

This PR intentionally excludes key exchanges based on EC-JPAKE; this will be covered in:
- #7280 
- #7281

Resolves #7255

## Gatekeeper checklist

- [x] **changelog** not required
- [x] **backport** not required
- [x] **tests** included

### CODE REVIEW

**Comment by valeriosetti** [comment]
The CI encountered only 1 error, but it seems to me that it is related to the CI itself, not to the code. Therefore I'm removing the "need-ci" label in order to start the review process

File: tests/include/test/drivers/crypto_config_test_driver_extension.h [hunk 1]
```diff
@@ -253,6 +253,7 @@
 
 #if defined(MBEDTLS_PSA_ACCEL_ALG_ECDSA)
 #if defined(MBEDTLS_PSA_ACCEL_ALG_ECDH)
+#if defined(MBEDTLS_PSA_ACCEL_ALG_JPAKE)
```
Comment (mpg) [inline tests/include/test/drivers/crypto_config_test_driver_extension.h:256]: Minor, pre-existing: I find `#if a && b && c` more readable than `#if a #if b #if c`.

**Review by mpg — DISMISSED** [review]
LGTM, just one minor point.

Looks like the follow-ups are going to be:
1. Enable `ssl-opt.sh` in reference and driver component with parity.
2. Re-enable the EC J-PAKE key exchange (still with parity).

If you agree with this list, can you take care of creating the issues?

**Comment by valeriosetti** [comment]
> If you agree with this list, can you take care of creating the issues?

Sure! I will take care of them ;)

**Review by mpg — APPROVED** [review]

**Review by AndrzejKurek — APPROVED** [review]

File: tests/scripts/all.sh [hunk 2]
```diff
@@ -2386,8 +2386,7 @@ component_test_psa_crypto_config_accel_ecjpake_use_psa () {
 
     # Build the main library
     loc_accel_flags="$loc_accel_flags $( echo "$loc_accel_list" | sed 's/[^ ]* */-DMBEDTLS_PSA_ACCEL_&/g' )"
-    #make CFLAGS="$ASAN_CFLAGS -O -Werror -I../tests/include -I../tests -I../../tests -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_TEST_LIBTESTDRIVER1 $loc_accel_flags" LDFLAGS="-ltestdriver1 $ASAN_CFLAGS"
-    make CFLAGS="$ASAN_CFLAGS -O0 -g -Werror -I../tests/include -I../tests -I../../tests -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_TEST_LIBTESTDRIVER1 $loc_accel_flags" LDFLAGS="-ltestdriver1 $ASAN_CFLAGS"
+    make CFLAGS="$ASAN_CFLAGS -O -Werror -I../tests/include -I../tests -I../../tests -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_TEST_LIBTESTDRIVER1 $loc_accel_flags" LDFLAGS="-ltestdriver1 $ASAN_CFLAGS"
```
Comment (mpg) [inline tests/scripts/all.sh:2389]: For the record: `ASAN_CFLAGS` includes `-O2` and I don't see a reason to change that to a lower optimization level here. It also includes `-Werror` btw, so we don't need to use it here either.

Not worth changing, especially as I'm hoping to merge the three components for ECDH, ECDSA, and ECJPAKE when wrapping up this EPIC. Just mentioning FYI.

File: tests/scripts/analyze_outcomes.py [hunk 3]
```diff
@@ -195,6 +195,18 @@ def do_analyze_driver_vs_reference(outcome_file, args):
             }
         }
     },
+    'analyze_driver_vs_reference_ecjpake': {
+        'test_function': do_analyze_driver_vs_reference,
+        'args': {
+            'component_ref': 'test_psa_crypto_config_reference_ecjpake_use_psa',
+            'component_driver': 'test_psa_crypto_config_accel_ecjpake_use_psa',
+            'ignored_suites': [
+                'ecjpake', # the software implementation that's excluded
```
[the remark this answers is not in the document]
Comment (mpg) [reply tests/scripts/analyze_outcomes.py:204]: I'm not sure I understand the question, can you clarify? All the (known) remaining work is tracked in [this EPIC](https://github.com/orgs/Mbed-TLS/projects/1?card_filter_query=is%3Aopen#column-19369852), and issues tend to be prefixed with "driver-only ECxxx" to reflect which area they're about.

File: tests/scripts/all.sh (hunk 2 above)
Comment (AndrzejKurek) [reply tests/scripts/all.sh:2389]: Sure, I just followed calls from other components that are building the library. Improvement area: we could define this long command somewhere to reuse it, there are currently 5 places that use it.

Comment (valeriosetti) [reply tests/scripts/all.sh:2389]: Yes, sorry for those changes, they are all connected to my debug sessions: being able to go through each line of the code without optimizations makes life easier for me. I must pay more attention when I add things for the commit

File: tests/scripts/analyze_outcomes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AndrzejKurek) [reply tests/scripts/analyze_outcomes.py:204]: Ah, nevermind this comment - I misunderstood the component pair that's tested here.

**Pull request merged**